client/src/main/java/io/pravega/client/admin/impl/ReaderGroupManagerImpl.java

```diff
@@ -105,10 +95,28 @@ public void deleteReaderGroup(String groupName) {
         StateSynchronizer<ReaderGroupState> synchronizer = clientFactory.createStateSynchronizer(NameUtils.getStreamForReaderGroup(groupName),
                 new ReaderGroupStateUpdatesSerializer(), new ReaderGroupStateInitSerializer(), SynchronizerConfig.builder().build());
         synchronizer.fetchUpdates();
-        UUID groupId = synchronizer.getState().getConfig().getReaderGroupId();
-        long generation = synchronizer.getState().getConfig().getGeneration();
-        getAndHandleExceptions(controller.deleteReaderGroup(scope, groupName, groupId, generation),
-                RuntimeException::new);
+        ReaderGroupConfig syncConfig = synchronizer.getState().getConfig();
+        if (ReaderGroupConfig.DEFAULT_UUID.equals(syncConfig.getReaderGroupId())
+            && ReaderGroupConfig.DEFAULT_GENERATION == syncConfig.getGeneration()) {
+            // migrate RG case
+            try {
+                controller.getReaderGroupConfig(scope, groupName)
+                        .thenCompose(conf -> controller.deleteReaderGroup(scope, groupName,
+                                conf.getReaderGroupId(), conf.getGeneration())).join();
+            } catch (IllegalArgumentException ex) {
+                controller.sealStream(scope, getStreamForReaderGroup(groupName))
+                          .thenCompose(b -> controller.deleteStream(scope, getStreamForReaderGroup(groupName)))
+                          .exceptionally(e -> {
+                                log.warn("Failed to delete stream", e);
```

Comment:
should we log that we failed to delete the readergroup?

Reply:
Yes, will add that and also add the cause that we failed to delete the Stream